func/mpi/CMakeLists.txt

```diff
@@ -3,15 +3,37 @@ set(FAASM_USER mpi)
 set(OMPI_CXX ${CXX})
 set(OMPI_C ${C})
 
+function(lulesh_mpi_target)
+    set(lulesh_exec lulesh_mpi)
+
+    if (NOT FAASM_BUILD_TYPE STREQUAL "wasm")
+        message("Native compilation of ${lulesh_exec} target is handled by the cmake file in the LULESH repo")
+        return()
+    endif ()
+
+    add_definitions("-DUSE_MPI=1")
+
+    set(lulesh_files
+            ../../third-party/LULESH/lulesh-comm.cc
+            ../../third-party/LULESH/lulesh-init.cc
+            ../../third-party/LULESH/lulesh-util.cc
+            ../../third-party/LULESH/lulesh.cc
+            )
+    add_executable(${lulesh_exec} ${lulesh_files})
+    target_link_libraries(${lulesh_exec} faasm faasmpi dlmalloc)
+
+    #set(ALL_OMP_FUNCS ${ALL_OMP_FUNCS} ${lulush_exec} PARENT_SCOPE)
+endfunction(lulesh_mpi_target)
```

Comment:
It looks like this is building lulesh files, hence should be added as a new target in the lulesh fork rather than in Faasm. It's also repeated over two files which would be nice to avoid.

I'd suggest having a look at the `compile_eigen` target to see how I'd recommend compiling a CMake project. It should be relatively straightforward to put this stuff into the Lulesh fork, then call `cmake` on it with the Faasm toolchain file. You can still include `target_link_libraries(xxx faasm faasmpi dlmalloc)` within the Lulesh CMake files, as these exist in the Faasm toolchain's sysroot